packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -31,20 +31,22 @@ function ensureSlash(path, needsSlash) {
   }
 }
 
-// We use "homepage" field to infer "public path" at which the app is served.
+// We use `PUBLIC_URL` environment variable or "homepage" field to infer
+// "public path" at which the app is served.
 // Webpack needs to know it to put the right <script> hrefs into HTML even in
 // single-page apps that may serve index.html for nested URLs like /todos/42.
 // We can't use a relative path in HTML because we don't want to load something
 // like /todos/42/static/js/bundle.7289d.js. We have to know the root.
-var homepagePath = require(paths.appPackageJson).homepage;
+var homepagePath = paths.publicUrl;
 var homepagePathname = homepagePath ? url.parse(homepagePath).pathname : '/';
+var servedPath = process.env.PUBLIC_URL ? homepagePath : homepagePathname;
```

Comment:
Can we avoid duplication in reading `process.env`? I feel like this may cause subtle bugs. Ideally the source of truth should be in `config/paths` and we should derive everything based on it. It's fine if it also calculates `servedPath`.